switchtec_dma.c

```diff
@@ -248,6 +248,7 @@ struct switchtec_dma_chan {
 	struct dma_chan dma_chan;
 	struct chan_hw_regs __iomem *mmio_chan_hw;
 	struct chan_fw_regs __iomem *mmio_chan_fw;
+	spinlock_t hw_ctrl_lock;
```

Comment:
Could use a description of why this lock is necessary....

I'd also probably reverse the patches instead of adding pause/resume functions only to change the locking in them.